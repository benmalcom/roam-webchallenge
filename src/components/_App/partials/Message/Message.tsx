import React from 'react';
import './Message.scss';
interface ComponentProps {
	error?: boolean;
	message: string;
}
export default ({error, message}: ComponentProps) => (
	<div className={`message mx-2 col-md-8 text-center text-muted alert alert-${error ? 'dark' : 'info'}`}>
		{message}
	</div>);
