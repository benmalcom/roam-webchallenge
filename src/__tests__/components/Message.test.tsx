import React from 'react';
import Message from '../../components/_App/partials/Message/Message';
import { mount } from 'enzyme';
describe('<Message />', () => {
	let props;
	let component;

	beforeEach(() => {
		props = {
			message: 'message',
			error: false,
		};
	});
	test('Component renders properly with props',  () => {
		component = mount(<Message {...props}/>);
		expect(component).toMatchSnapshot();
		expect(component.prop('message')).toEqual(props.message);
		expect(component.prop('error')).toBeFalsy();
	});
});
