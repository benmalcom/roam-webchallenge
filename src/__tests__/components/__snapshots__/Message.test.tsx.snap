// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message /> Component renders properly with props 1`] = `
<Component
  error={false}
  message="message"
>
  <div
    className="message mx-2 col-md-8 text-center text-muted alert alert-info"
  >
    message
  </div>
</Component>
`;
